tests/test_parser.py

```diff
@@ -175,6 +220,35 @@ def test_with_unknown_args(self):
         self.assertEqual(args.gmp_password, 'foo')
         self.assertEqual(script_args, ['--bar', '--bar2'])
 
+    @mock.patch('gvmtools.parser.logging')
+    def test_socket_has_no_timeout(
+        self, logging_mock
+    ):  # pylint: disable=unused-argument
+        # pylint: disable=protected-access
+        self.parser._parser = mock.MagicMock()
+        args_mock = mock.MagicMock()
+        args_mock.timeout = -1
+        self.parser._parser.parse_known_args = mock.MagicMock(
+            return_value=(args_mock, mock.MagicMock())
+        )
+
+        args, _ = self.parser.parse_known_args(
+            ['socket', '--timeout', '--', '-1']
+        )
+
+        self.assertIsNone(args.timeout)
```

Comment:
I don't understand this test at the moment. For me the intention is not clear and I don't understand why you are mocking the parser completely to test the parser. Maybe you can explain it to me and/or add some comments/hints to the code?

Reply:
The problem I was facing there is if I pass "-1" to parse_known_args, it is interpreted as an argument and not as a negative number. I could not figure out how I can pass -1 and get timeout set to the actual number -1. That's why I have mocked the method.

Reply:
NVM, mocking the parser was obsolete. I have removed that code.